gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/oneconvergence_service_node_driver.py

```diff
@@ -467,47 +469,187 @@ def update_policy_target_removed(self, context, policy_target):
         if service_type != pconst.LOADBALANCER:
             return
 
-        self.update(context)
+        self.update(context, pt_added_or_removed=True)
 
     @log.log
     def notify_chain_parameters_updated(self, context):
         pass # We are not using the classifier specified in redirect Rule
 
     @log.log
+    def check_for_existing_service(self, context):
+        if context._provider_group['provided_policy_rule_sets']:
+            provided_policy_rule_set_id = context._provider_group[
+                'provided_policy_rule_sets'][0]
+        else:
+            return True, None, True
+        provided_policy_rule_set = context._gbp_plugin.get_policy_rule_set(
+            context.plugin_context, provided_policy_rule_set_id)
+        cons_policy_target_groups = provided_policy_rule_set[
+            'consuming_policy_target_groups']
+
+        stack, sc_instances = self.get_firewall_stack_id(context,
+                                             cons_policy_target_groups)
+
+        if context.consumer['id'] in cons_policy_target_groups:
+            cons_policy_target_groups.remove(context.consumer['id'])
+
+        if stack:
+            return True, cons_policy_target_groups, False
+        else:
+             return False, cons_policy_target_groups, False
+        #
+        # if cons_policy_target_groups:
+        #     return True, cons_policy_target_groups, False
+        # else:
+        #     return False, cons_policy_target_groups, False
+
+    @log.log
+    def append_firewall_rule(self, context, stack_template, provider_cidr,
+                             consumer_cidr, fw_template_properties):
+        resources_key = fw_template_properties['resources_key']
+        properties_key = fw_template_properties['properties_key']
+        fw_rule_keys = fw_template_properties['fw_rule_keys']
+        rule_name = fw_template_properties['name']
+        fw_policy_key = fw_template_properties['fw_policy_key']
+        i = 1
```

Comment:
Wont the rules be overwritten for subsequent consumers since the value of i always starts from 1 ?